Policy violation context for Ansible Automation Platform (#4538)
open-cluster-management/rhacm-docs: governance/manage_policy_intro.adoc

- _Run everyEvent mode_: When a policy is violated, the automation runs every time for each unique policy violation per managed cluster. Use the `DelayAfterRunSeconds` parameter to set the minimum seconds before an automation can be restarted on the same cluster. If the policy is violated multiple times during the delay period and kept in the violated state, the automation runs one time after the delay period. The default is 0 seconds and is only applicable for the `everyEvent` mode. When you run `everyEvent` mode, the extra variable of `target_clusters` and {aap-short} Job template is the same as _once mode_.
- _Disable automation_: When the scheduled automation is set to `disabled`, the automation does not run until the setting is updated.

> - The following variables are automatically provided in the `extra_vars` of the {aap-short} Job:

Reviewer: ```suggestion

The following variables are automatically provided in the `extra_vars` of the {aap-short} Job:
```

- _Run everyEvent mode_: When a policy is violated, the automation runs every time for each unique policy violation per managed cluster. Use the `DelayAfterRunSeconds` parameter to set the minimum seconds before an automation can be restarted on the same cluster. If the policy is violated multiple times during the delay period and kept in the violated state, the automation runs one time after the delay period. The default is 0 seconds and is only applicable for the `everyEvent` mode. When you run `everyEvent` mode, the extra variable of `target_clusters` and {aap-short} Job template is the same as _once mode_.
- _Disable automation_: When the scheduled automation is set to `disabled`, the automation does not run until the setting is updated.

The following variables are automatically provided in the `extra_vars` of the {aap-short} Job:

> * `policy_name`: The name of the non-compliant root policy that intiates the {aap-short} job on the hub cluster.

Reviewer: @dockerymick I just noticed a typo here:
`intiates` => `initiates`

[#grc-automation-configuration]
== Governance automation configuration

If there is a configured automation for a specific policy, you can select the automation to view more details. View the following descriptions of the schedule frequency options for your automation:

- _Manual run_: Manually set this automation to run once. After the automation runs, it is set to `disabled`. *Note:* You can only select _Manual run_ mode when the schedule frequency is disabled.
- _Run once mode_: When a policy is violated, the automation runs one time. After the automation runs, it is set to `disabled`. After the automation is set to `disabled`, you must continue to run the automation manually. When you run _once mode_, the extra variable of `target_clusters` is automatically supplied with the list of clusters that violated the policy. The {aap-short} Job template must have `PROMPT ON LAUNCH` enabled for the `EXTRA VARIABLES` section (also known as `extra_vars`).

> - View the following parameters listed in the `extra_vars` section:

Reviewer: Could you please move this to below all the modes?

> - View the following parameters listed in the `extra_vars` section:

Reviewer: ```suggestion
- The following variables are automatically provided in the `extra_vars` of the {aap-short} Job:
```

[#grc-automation-configuration]
== Governance automation configuration

If there is a configured automation for a specific policy, you can select the automation to view more details. View the following descriptions of the schedule frequency options for your automation:

- _Manual run_: Manually set this automation to run once. After the automation runs, it is set to `disabled`. *Note:* You can only select _Manual run_ mode when the schedule frequency is disabled.
- _Run once mode_: When a policy is violated, the automation runs one time. After the automation runs, it is set to `disabled`. After the automation is set to `disabled`, you must continue to run the automation manually. When you run _once mode_, the extra variable of `target_clusters` is automatically supplied with the list of clusters that violated the policy. The {aap-short} Job template must have `PROMPT ON LAUNCH` enabled for the `EXTRA VARIABLES` section (also known as `extra_vars`).
+

> ** Use the following parameters in the `extra_vars` section to receive additional policy violation details:

Reviewer: The feature actually provides these variables in `extra_vars` automatically.

Reviewer: These values are also not specific to run once mode. The other modes also have this information.

+
* `policy_name`: The name of the non-compliant root policy that intiates the {aap-short} job on the hub cluster.
* `namespace`: The namespace of the root policy. 

> * `hub_cluster`: The name of the hub cluster, which is received from the Kubernetes configuration host value.

Reviewer: ```suggestion
* `hub_cluster`: The name of the hub cluster, which is determined by the value in the `clusters` `DNS` object.
```

* `policy_set`: This parameter contains all associated policy set names of the root policy. If the policy is not within a policy set, the `policy_set` parameter is empty.

> * `policy_violation_context`: This parameter contains a list of non-compliant cluster names, and the value is the raw replicated policy status for each non-compliant cluster. The policy automation checks the hub cluster namespaces for replicated policies, then filters the non-compliant, replicated policies.

Reviewer: There was too many technical details.
```suggestion
* `policy_violation_context`: This parameter contains a list of non-compliant cluster names, and the value is the policy `status` field for each non-compliant cluster.
```
